ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh

```diff
@@ -117,118 +144,209 @@ metadata:
+# - There will be a failed systemd service: create-local-etcd.service
 wait_for_machineconfig_done
+check_etcd_unmounted_on_local_disk
+cp_machines=$(oc get machines -n openshift-machine-api --selector='machine.openshift.io/cluster-api-machine-role=master' --no-headers -o custom-columns=NAME:.metadata.name)
+for machine in ${cp_machines}; do
+  if oc get machine -n openshift-machine-api "${machine}" -o jsonpath='{.spec.providerSpec.value}' | grep -q 'additionalBlockDevices'; then
+    info "ERROR: Machine ${machine} does has the etcd block device"
+    exit 1
+  fi
+  info "INFO: Machine ${machine} does not have the etcd block device"
+done
+info 'INFO: All control plane machines do not have the etcd block device'
+cp_nodes=$(oc get nodes -l node-role.kubernetes.io/master --no-headers -o custom-columns=NAME:.metadata.name)
+for node in ${cp_nodes}; do
+  info "INFO: Checking create-local-etcd.service on node ${node}"
+  service=$(oc debug node/"${node}" -- chroot /host systemctl status create-local-etcd.service)
+  if [[ "${service}" == *"Active: failed"* ]]; then
+    info "INFO: create-local-etcd.service failed on node ${node}"
+  else
+    info "ERROR: create-local-etcd.service did not fail on node ${node}"
+    exit 1
+  fi
+done
+
+info "This tests that it is possible to remove the local etcd configuration when it is not in use."
+# Preparation:
+#
+# - Remove MachineConfig
+info "INFO: Remove 98-var-lib-etcd MachineConfig"
+oc delete -f /tmp/98-var-lib-etcd.yaml
+
+# Verify:
+# 
+# - The cluster is operating correctly
+# - There are no additional failed systemd services
+wait_for_machineconfig_done
+cp_nodes=$(oc get nodes -l node-role.kubernetes.io/master --no-headers -o custom-columns=NAME:.metadata.name)
+for node in ${cp_nodes}; do
+  info "INFO: Checking create-local-etcd.service on node ${node}"
+  service=$(oc debug node/"${node}" -- chroot /host systemctl status create-local-etcd.service)
+  if [[ "${service}" == *"Active: failed"* ]]; then
+    info "ERROR: create-local-etcd.service failed on node ${node}"
+    exit 1
+  else
+    info "INFO: create-local-etcd.service did not fail on node ${node}"
+  fi
+done
+
+info "This tests that the MachineConfig works correctly when deployed on a new machine without any existing etcd state to migrate."
+# Preparation:
+#
+# - Apply MachineConfig again
+# - Wait for rollout to complete
+# - Update CPMS to add local etcd
+# - Wait for rollout to complete
+info "INFO: Apply 98-var-lib-etcd MachineConfig"
+oc create -f /tmp/98-var-lib-etcd.yaml
+wait_for_machineconfig_done
+info "INFO: Adding additionalBlockDevices to the control plane machineset"
+oc patch ControlPlaneMachineSet/cluster -n openshift-machine-api --type json -p '[{"op": "add", "path": "/spec/template/machines_v1beta1_machine_openshift_io/spec/providerSpec/value/additionalBlockDevices", "value": [{"name": "etcd", "sizeGiB": 10, "storage": {"type": "Local"}}]}]'
```

Comment:
here we also need to re-change the flavor to use one that have an ephemeral disk